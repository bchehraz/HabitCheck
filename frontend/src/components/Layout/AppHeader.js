import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

const Container = styled.div`
  margin: 0 auto;
  padding: 2rem;
  background: white;
  user-select: none;

  @media only screen and (min-width: 768px) {
    p {
      font-size: 1.2em;
    }

    h1 {
      font-size: 3em;
    }
  }
`

const getDate = () => {
  const days = [
    "Sunday",
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday",
  ]
  const months = [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December",
  ]

  const date = new Date()

  return `${days[date.getDay()]}, ${
    months[date.getMonth()]
  } ${date.getDate()} ${date.getFullYear()}`
}

const AppHeader = ({ title }) => (
  <header>
    <Container>
      <h1>{title}</h1>
      <p>{getDate()}</p>
    </Container>
  </header>
)

AppHeader.propTypes = {
  title: PropTypes.string.isRequired,
}

AppHeader.defaultProps = {
  siteTitle: ``,
}

export default AppHeader
