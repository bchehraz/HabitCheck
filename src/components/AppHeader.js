import React from "react"
import { Link } from "gatsby"
import PropTypes from "prop-types"
import styled from 'styled-components';

const Container = styled.div`
  margin: 0 auto;
  padding: 1.45rem 1.0875rem;
  background: white;
`;

const getDate = () => {
  const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
  const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

  const date = new Date();

  return `${days[date.getDay()]}, ${months[date.getMonth()]} ${date.getDate()} ${date.getFullYear()}`;
}

const AppHeader = ({ title }) => (
  <header>
    <Container>
      <h1>
        {title}
      </h1>
      <p>{getDate()}</p>
    </Container>
  </header>
)

AppHeader.propTypes = {
  title: PropTypes.string.isRequired,
}

AppHeader.defaultProps = {
  siteTitle: ``,
}

export default AppHeader;
