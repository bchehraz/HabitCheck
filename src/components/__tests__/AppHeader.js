import React from "react"
import renderer from "react-test-renderer"

import AppHeader from "../AppHeader"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<AppHeader title="HabitCheck" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
