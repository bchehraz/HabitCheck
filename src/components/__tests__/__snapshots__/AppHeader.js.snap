// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <div
    className="sc-bdVaJa bzVTIk"
  >
    <h1>
      HabitCheck
    </h1>
    <p>
      Wednesday, January 22 2020
    </p>
  </div>
</header>
`;
